src/conversations/Conversation.ts

```diff
@@ -1,43 +1,62 @@
-import Stream from '../Stream'
+import Stream, { noTransformation } from '../Stream'
 import Client, {
   ListMessagesOptions,
   ListMessagesPaginatedOptions,
   SendOptions,
 } from '../Client'
-import Message from '../Message'
+import {
+  InvitationContext,
+  InvitationV1,
+  SealedInvitationHeaderV1,
+} from '../Invitation'
+import { MessageV1, MessageV2 } from '../Message'
+import { messageApi, xmtpEnvelope } from '@xmtp/proto'
+import { encrypt, decrypt, SignedPublicKey, Signature } from '../crypto'
+import Ciphertext from '../crypto/Ciphertext'
+import { sha256 } from '../crypto/encryption'
+import { dateToNs, nsToDate } from '../utils'
 
 /* eslint-disable @typescript-eslint/explicit-module-boundary-types */
 
 /**
  * Conversation class allows you to view, stream, and send messages to/from a peer address
  */
-export default class Conversation {
-  peerAddress: string
+export class ConversationV1 {
+  private _peerAddress: string
+  sent: Date
   private client: Client
 
-  constructor(client: Client, address: string) {
-    this.peerAddress = address
+  constructor(client: Client, address: string, sent: Date) {
+    this._peerAddress = address
     this.client = client
+    this.sent = sent
```

Comment:
What is the `sent` field used for?

Reply:
To sort Conversations. Currently we sort by peerAddress, that's not going to be sufficient going forward.

Reply:
https://github.com/xmtp/xmtp-js/blob/main/src/conversations/Conversations.ts#L43-L47

Reply:
Makes sense for the ordering.

Maybe it's just the name that's bothering me. I could imagine `conversation.sent` could be confusing for developers (is it the time the last message was sent? The time the first message was sent?). In reality, it's neither. For V1 it is the time that the first message was sent in the most recent session of either user, which is a really arbitrary and useless timestamp.

`createdAt` would be better, but it's not accurate for V1 since we are actually using the latest `intro` and not the first.

Reply:
What if we use the earliest seen timestamp instead of the latest seen timestamp in `getIntroductionPeers`. That would let us call it something like `createdAt` and would be a more meaningful value.

Reply:
I think that makes sense. Do you want to make that change?

Reply:
Can do